data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/SearchAccessorStrategy.java

```diff
@@ -274,18 +300,107 @@ private SSLContext getTrustAllStrategy() {
-                    "The version of the OpenSearch cluster passed is %s", OPENSEARCH_POINT_IN_TIME_SUPPORT_VERSION_CUTOFF, version));
+                    "The version of the OpenSearch cluster passed is %s. Elasticsearch clusters with build-flavor %s do not support point in time",
+                            distribution.startsWith(ELASTICSEARCH_DISTRIBUTION) ? ELASTICSEARCH_POINT_IN_TIME_SUPPORT_VERSION_CUTOFF : OPENSEARCH_POINT_IN_TIME_SUPPORT_VERSION_CUTOFF,
+                            version, ELASTICSEARCH_OSS_BUILD_FLAVOR));
         }
     }
 
-    private boolean versionSupportsPointInTimeForOpenSearch(final String version) {
-        final DefaultArtifactVersion cutoffVersion = new DefaultArtifactVersion(OPENSEARCH_POINT_IN_TIME_SUPPORT_VERSION_CUTOFF);
+    private boolean versionSupportsPointInTime(final String distribution, final String version) {
         final DefaultArtifactVersion actualVersion = new DefaultArtifactVersion(version);
+
+        DefaultArtifactVersion cutoffVersion;
+        if (distribution.startsWith(ELASTICSEARCH_DISTRIBUTION)) {
+            if (distribution.endsWith(ELASTICSEARCH_OSS_BUILD_FLAVOR)) {
+                return false;
+            }
+            cutoffVersion = new DefaultArtifactVersion(ELASTICSEARCH_POINT_IN_TIME_SUPPORT_VERSION_CUTOFF);
+        } else {
+            cutoffVersion = new DefaultArtifactVersion(OPENSEARCH_POINT_IN_TIME_SUPPORT_VERSION_CUTOFF);
+        }
         return actualVersion.compareTo(cutoffVersion) >= 0;
     }
+
+    private Pair<String, String> getDistributionAndVersionNumber(final InfoResponse infoResponseOpenSearch, final ElasticsearchClient elasticsearchClient) {
+        if (Objects.nonNull(infoResponseOpenSearch)) {
+            return Pair.of(infoResponseOpenSearch.version().distribution(), infoResponseOpenSearch.version().number());
+        }
+
+        try {
+            final co.elastic.clients.elasticsearch.core.InfoResponse infoResponseElasticsearch = elasticsearchClient.info();
+            return Pair.of(ELASTICSEARCH_DISTRIBUTION + "-" + infoResponseElasticsearch.version().buildFlavor(), infoResponseElasticsearch.version().number());
+        } catch (final Exception e) {
+            throw new RuntimeException("Unable to call info API using the elasticsearch client", e);
+        }
+    }
+
+    private void validateDistribution(final String distribution) {
+        if (!distribution.equals(OPENSEARCH_DISTRIBUTION) && !distribution.startsWith(ELASTICSEARCH_DISTRIBUTION) && !distribution.equals(OPENDISTRO_DISTRIUBTION)) {
+            throw new IllegalArgumentException(String.format("Only %s, %s, or %s distributions are supported at this time. The cluster distribution being used is '%s'",
+                    OPENSEARCH_DISTRIBUTION, OPENDISTRO_DISTRIUBTION, ELASTICSEARCH_DISTRIBUTION, distribution));
+        }
+    }
+
+    private ElasticsearchTransport createElasticSearchTransport(final org.elasticsearch.client.RestClient restClient) {
```

Comment:
The client construction logic here and below belongs in another class

Reply:
Will make a helper factory class and move the functions for client creation to that class for opensearch and elasticsearch